From 2ebbc222603f92769b60412c4592e3fb2d6c8d00 Mon Sep 17 00:00:00 2001
From: Jonathan Kang <jonathankang@gnome.org>
Date: Thu, 1 Sep 2022 10:15:15 +0800
Subject: [PATCH] zypp: Don't refresh repos before searching

Refreshing repos should be done by calling "refresh-cache", not inside
backend_find_packages_thread.

This fixes the issue that GNOME Software calls multiple search-files
methods which takes very long time to finish. As most of the time was
spent refreshing repos multiple times.

https://bugzilla.suse.com/show_bug.cgi?id=1202796
---
 backends/zypp/pk-backend-zypp.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/backends/zypp/pk-backend-zypp.cpp b/backends/zypp/pk-backend-zypp.cpp
index 62adbb8be..c87972079 100644
--- a/backends/zypp/pk-backend-zypp.cpp
+++ b/backends/zypp/pk-backend-zypp.cpp
@@ -3109,11 +3109,6 @@ backend_find_packages_thread (PkBackendJob *job, GVariant *params, gpointer user
 		return;
 	}
 
-	// refresh the repos before searching
-	if (!zypp_refresh_cache (job, zypp, FALSE)) {
-		return;
-	}
-
 	search = values[0];  //Fixme - support the possible multiple values (logical OR search)
 	role = pk_backend_job_get_role(job);
 
-- 
2.37.1

